source/Mission.cpp

```diff
@@ -1148,57 +1148,60 @@ bool Mission::HasShip(const shared_ptr<Ship> &ship) const
 // about it. This may affect the mission status or display a message.
 void Mission::Do(const ShipEvent &event, PlayerInfo &player, UI *ui)
 {
-	if(event.TargetGovernment()->IsPlayer() && !hasFailed)
+	if(!HasFailed(player))
```

Comment:
to avoid compiler complaints after syncing with master
```suggestion
	if(!IsFailed(player))
```